### DISCUSSION TITLE
Csla based objects and ReportDataSource class

### DISCUSSION THREAD

**Discussion by Chicagoan2016** [MarimerLLC/csla#3370] [Questions]
We have an application with one UI in Winforms.
I tried to use ReportDataSource directly with Csla object but it complained about not supplying a datasource that implements Type, IEnumerable etc. :)
We have ObjectAdapter in Csla so I populated a dataset from a Csla based object and we are good but I am curious, is it possible to use Csla object without ObjectAdapter (as DataSource for rdlc report) ?

Kind regards

**Comment by danielmartind** [comment]
I've always used a ReadOnlyList with my reports. Are you using a BindingSource?

**Reply by Chicagoan2016** [reply]
We usually have rdlc reports (SSRS).
and we set DataSource 
```cs
   reportEngine.DataSources.Add(new ReportDataSource(ds.DataSetName, ds.Tables[0]));
```

**Reply by danielmartind** [reply]
I load my report data using a read only list and set the as the data source to BindingSource component. That is passed to the ReportDataSource. Mostly configured in the designer in the past but a bit hit and miss these days.

`ReportViewer1.LocalReport.DataSources.Add(new ReportDataSource("DataSet", BindingSource));`

`BindingSource.DataSource = ReadOnlyList.GetReadOnlyList()`
